### PR TITLE
Fix cpfd config attribute

### PR DESCRIPTION
**Pull request by evading**
The `cfg(all(...))` gating for the cpfd module, added in #114, is not working for ESP-IDF versions > 5.2.
This patch introduces a new cfg, `cpfd`, that is set from the build script if ESP-IDF version is above 5.2.1. The new cfg is then used to conditionally include the cpfd support.

### CODE REVIEW

File: src/server/ble_characteristic.rs [hunk 1]
```diff
@@ -21,11 +17,7 @@ use crate::{
 };
 
 cfg_if::cfg_if! {
-  if #[cfg(all(
-    esp_idf_version_major = "5",
-    esp_idf_version_minor = "2",
-    not(any(esp_idf_version_patch = "0", esp_idf_version_patch = "1", esp_idf_version_patch="2"))
-  ))] {
+  if #[cfg(cpfd)] {
```
Comment (taks) [inline src/server/ble_characteristic.rs:20]: This part is not related to cpfd.
The cpfd was changed in ESP-IDF version 5.2.1, but this part was changed in ESP-IDF version 5.2.3.

Comment (evading) [reply src/server/ble_characteristic.rs:20]: Oh. Sorry. I'll push a new patch asap.

Comment (evading) [reply src/server/ble_characteristic.rs:20]: Done.

File: build.rs [hunk 2]
```diff
@@ -12,4 +12,32 @@ fn main() {
...
+  let esp_idf_minor = extract_version_component(esp_idf_info.args.iter(), "esp_idf_version_minor");
+  let esp_idf_patch = extract_version_component(esp_idf_info.args.iter(), "esp_idf_version_patch");
+
+  println!("cargo::rustc-check-cfg=cfg(cpfd)");
+  if esp_idf_major > 5
+    || (esp_idf_major == 5 && esp_idf_minor > 2)
+    || (esp_idf_major == 5 && esp_idf_minor == 2 && esp_idf_patch > 0)
+  {
+    println!("cargo::rustc-cfg=cpfd");
+  }
```
Comment (taks) [inline build.rs:42]: This can be simplified as follows
```rust
if (esp_idf_major, esp_idf_minor, esp_idf_patch) > (5, 2, 0) {
  println!("cargo::rustc-cfg=cpfd");
}
```

Comment (evading) [reply build.rs]: Nice! TIL
I updated the patch a bit and used that feature of `Vec`. Let me know what you think.

**Comment by taks** [comment]
Thanks.
